storage/user.go

```diff
@@ -0,0 +1,70 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+type UserStore interface {
+	InsertUser(context.Context, *User) error
+	UpdateUser(context.Context, *User) error
+	SelectUser(context.Context, string) (*User, error)
+}
+
+func NewUserStore(db *sql.DB) UserStore {
+	return &userStore{
+		db: db,
+	}
+}
+
+type userStore struct {
+	db *sql.DB
+}
+
+type User struct {
+	Nick        string
+	PwHash      string
+	Description string
+}
+
+func (s *userStore) InsertUser(ctx context.Context, u *User) error {
```

Comment:
Context passed here could be used in some way, see this [example in Go Playground](https://go.dev/play/p/MylTMODZHIT).

Context from request could be wrapped in `WithTimeout` before. Then errors from context could be checked in the beginning of db functions.

Real life use case is when the response has some time limit and many potentially long functions have to be called to prepare the response. If the execution of first function already exceeds the deadline, there is no point in continuing the work - by checking the context error we can return early.
```suggestion
func (s *userStore) InsertUser(ctx context.Context, u *User) error {
	if err := ctx.Err(); err != nil {
		return err
	}
```